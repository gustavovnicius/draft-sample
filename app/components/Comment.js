import React, { Component } from 'react'
import { InlineForm } from 'rebass'

const styles = {
}

class Comment extends Component {
  render() {
    if (this.props.comment.saved) {
      return this.renderSavedComment()
    }

    return this.renderUnsavedComment()
  }

  renderSavedComment() {

  }

  renderUnsavedComment() {
    return (
      <div>
        <InlineForm
          label="Write a comment"
          placeholder="Write a comment"
          name="comment"
          buttonLabel="Save"
        />
      </div>
    )
  }
}

export default Comment
