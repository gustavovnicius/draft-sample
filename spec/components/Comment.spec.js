import React from 'react'
import renderer from 'react-test-renderer'
import Comment from 'components/Comment.js'

describe('Comment', () => {
  it('renders', () => {
    const comment = {
      data: {
        comment: '',
        saved: false
      },
      mutability: 'MUTABLE',
      type: 'COMMENT'
    }
    const app = renderer.create(
      <Comment comment={comment} />
    )
    const tree = app.toJSON()

    expect(tree).toMatchSnapshot()
  });
});
