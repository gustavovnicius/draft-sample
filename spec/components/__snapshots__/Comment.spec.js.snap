exports[`Comment renders 1`] = `
<div>
  <form
    className="InlineForm"
    style={
      Object {
        "alignItems": "center",
        "boxSizing": "border-box",
        "display": "flex",
      }
    }>
    <div
      className="Input"
      style={
        Object {
          "boxSizing": "border-box",
          "color": null,
          "flex": "1 1 auto",
          "marginBottom": 0,
        }
      }>
      <label
        className="Label"
        htmlFor="comment"
        style={
          Object {
            "boxSizing": "border-box",
            "clip": "rect(1px, 1px, 1px, 1px)",
            "fontSize": 14,
            "fontWeight": 600,
            "height": 1,
            "lineHeight": 1,
            "overflow": "hidden",
            "position": "absolute",
            "width": 1,
          }
        }>
        Write a comment
      </label>
      <input
        name="comment"
        onChange={[Function]}
        placeholder="Write a comment"
        style={
          Object {
            "backgroundColor": "rgba(255, 255, 255, .25)",
            "borderColor": "rgba(0, 0, 0, .25)",
            "borderRadius": "2px 0 0 2px",
            "borderStyle": "solid",
            "borderWidth": 1,
            "boxSizing": "border-box",
            "color": "inherit",
            "display": "block",
            "fontFamily": "inherit",
            "fontSize": "inherit",
            "height": 32,
            "margin": 0,
            "paddingLeft": 8,
            "paddingRight": 8,
            "width": "100%",
          }
        }
        type="text"
        value={undefined} />
    </div>
    <button
      className="ButtonOutline"
      href={undefined}
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "transparent",
          "border": 0,
          "borderRadius": "0 2px 2px 0",
          "boxShadow": "inset 0 0 0 1px",
          "boxSizing": "border-box",
          "color": "#08e",
          "cursor": "pointer",
          "display": "inline-block",
          "fontFamily": "inherit",
          "fontSize": 14,
          "fontWeight": 600,
          "lineHeight": "16px",
          "margin": 0,
          "marginLeft": -1,
          "minHeight": 32,
          "paddingBottom": 8,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 8,
          "textDecoration": "none",
        }
      }
      type="submit">
      Save
    </button>
  </form>
</div>
`;
